### ISSUE TITLE
[Bug Report] filtering out non File instances from v-file-input

### ISSUE THREAD

**Issue by matthieusieben** [vuetifyjs/vuetify#10306]
### Problem to solve
The [issue](https://github.com/vuetifyjs/vuetify/issues/10281) that is the origin of this commit contains an example which is clearly a bad use of the component.

On my platform, I use a form component which can persist data when the browser is refreshed.
However, `File` instances cannot ne created arbitrarily in any browsers (IE / Edge, of course!).

I was able to "emulate" previously uploaded files using a custom object with all the file info (name, size, ...). This was working perfectly.

The change from [ebdb4f64](https://github.com/vuetifyjs/vuetify/commit/ebdb4f64cdc3a1b118e61992f235c60b3586f168) removed this ability in order to support an invalid usage of the component.

My workaround:
```js
const CustomFileInput = VFileInput.extend({
  computed: {
    internalArrayValue() {
      // @hack: Allow non "File" instances to be used
      return asArray(this.internalValue)
    },
  },
})
```

### Proposed solution
Revert commit https://github.com/vuetifyjs/vuetify/commit/ebdb4f64cdc3a1b118e61992f235c60b3586f168

**Comment by jacekkarczmarczyk** [comment]
I agree with this, additinally imho invalid data should rather result in some error/warning instead of silently ignoring entries

**Comment by johnleider** [comment]
After talking with the team we are classifying this as a regression. It can be resolved without regressing #10281
